stm/stm.ml

```diff
@@ -152,7 +197,14 @@ let mkTransCmd cast cids ceff cqueue =
 let summary_pstate = [ Evarutil.meta_counter_summary_name;
                        Evd.evar_counter_summary_name;
                        "program-tcc-table" ]
-type cached_state =
+type cached_state = {
+  mutable parsing : Parser.state;
+    (* Mutable because in some cases we we can't produce it without executing
+     * the command, but execution needs to add a node to the dag, and to
+     * build a node we give the parsing state :-/ *)
```

Comment:
Well, in one case you assign the wrong state to the node, to correct if afterwards, whereas in the other it always holds that a node has the right state. So IMHO it is more about correctness than about taste, for instance in the first case you have to worry about errors in the whole path and invalidate, don't you?